sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -1145,6 +1147,54 @@ DataT fetch_image_array(const unsampled_image_handle &imageHandle
 #endif
 }
 
+/**
+ *  @brief   Fetch data from an unsampled cubemap image using its handle
+ *
+ *  @tparam  DataT The return type
+ *
+ *  @param   imageHandle The image handle
+ *  @param   coords The coordinates at which to fetch image data (int2 only)
+ *  @param   face The cubemap face at which to fetch
+ *  @return  Image data
+ */
+template <typename DataT>
+DataT fetch_cubemap(const unsampled_image_handle &imageHandle,
+                    const int2 &coords, const unsigned int face) {
+  return fetch_image_array<DataT>(imageHandle, coords, face);
+}
+
+/**
+ *  @brief   Sample a cubemap image using its handle
+ *
+ *  @tparam  DataT The return type
+ *
+ *  @param   imageHandle The image handle
+ *  @param   dirVec The direction vector at which to sample image data (float3
+ *           only)
+ *  @return  Image data
+ */
+template <typename DataT, typename HintT = DataT>
+DataT sample_cubemap(const sampled_image_handle &imageHandle [[maybe_unused]],
+                     const sycl::float3 &dirVec [[maybe_unused]]) {
+
+#ifdef __SYCL_DEVICE_ONLY__
+  if constexpr (detail::is_recognized_standard_type<DataT>()) {
+    return __invoke__ImageReadCubemap<DataT, uint64_t>(imageHandle.raw_handle,
+                                                       dirVec);
+  } else {
+    static_assert(sizeof(HintT) == sizeof(DataT),
+                  "When trying to read a user-defined type, HintT must be of "
+                  "the same size as the user-defined DataT.");
+    static_assert(detail::is_recognized_standard_type<HintT>(),
+                  "HintT must always be a recognized standard type");
+    return sycl::bit_cast<DataT>(__invoke__ImageReadCubemap<DataT, uint64_t>(
+        imageHandle.raw_handle, dirVec));
+  }
+#else
+  assert(false); // Bindless images not yet implemented on host
```

Comment:
I wonder if this should throw rather than asserting. If someone goes through this code path, they would not receive any error, nor even a warning, but the code would not do what they expect. Any thoughts?

Reply:
This is kernel code. I didn't think you could throw in kernel code. Am I wrong?

Reply:
A throw should work as it would only run on the host. So no worries about running it on a device that does not support exceptions. Though the sycl 2020 spec states that exceptions are not supported. So perhaps it is better to put something like `assert(false && "Bindless images not yet implemented on host");` instead.

Unless perhaps throws are used elsewhere in situations similar to this. I don't think this should be an outlier using a throw.

Reply:
Sorry, I wasn't clear in my first comment: the problem is that `assert` only runs in debug mode, so if a user is using production mode (the most frequent), the assertion wouldn't work, and the program would continue as if nothing was wrong.

Reply:
Yes, as long as it's being tracked, it works for me.